# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/xlockmore/hotfix-gfree.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

GLib 2.78+ defines g_free() as an unparenthesized ternary macro, so a (void) cast
binds to its condition: drop the redundant casts.

diff -Nurp xlockmore-5.89.orig/xglock/xglock.c xlockmore-5.89/xglock/xglock.c
--- xlockmore-5.89.orig/xglock/xglock.c	2026-06-30 04:04:54.000000000 +0200
+++ xlockmore-5.89/xglock/xglock.c	2026-09-27 19:18:49.376069968 +0200
@@ -666,7 +666,7 @@ color_selection_cancel_cb(GtkWidget *w,
     struct_option_fntcol_callback *cbs_ptr =(struct_option_fntcol_callback *)udata;
     gtk_widget_hide(cbs_ptr->fntcol_dialog);
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -729,7 +729,7 @@ color_selection_ok_cb(GtkWidget *w,
     gtk_entry_set_text( GTK_ENTRY(cbs_ptr->entry), colorstr );
     gtk_widget_hide( cbs_ptr->fntcol_dialog );
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -862,7 +862,7 @@ font_select_cancel_cb(GtkWidget *w,
     struct_option_fntcol_callback *cbs_ptr = (struct_option_fntcol_callback *)udata;
     gtk_widget_hide(cbs_ptr->fntcol_dialog);
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -879,7 +879,7 @@ font_select_ok_cb(GtkWidget *w,
     gtk_entry_set_text(GTK_ENTRY(cbs_ptr->entry),
                        gtk_font_selection_dialog_get_font_name(GTK_FONT_SELECTION_DIALOG(cbs_ptr->fntcol_dialog)));
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -931,7 +931,7 @@ file_selection_cancel_cb(GtkWidget *w,
     struct_option_fntcol_callback *cbs_ptr= (struct_option_fntcol_callback *)udata;
     gtk_widget_hide( cbs_ptr->fntcol_dialog );
     gtk_widget_destroy( cbs_ptr->fntcol_dialog );
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -954,7 +954,7 @@ file_selection_ok_cb(GtkWidget *w,
         gtk_widget_hide(cbs_ptr->fntcol_dialog);
         gtk_entry_set_text(GTK_ENTRY(cbs_ptr->entry), filename);
         gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-        (void) g_free(udata);
+        g_free(udata);
     }
 }
 
